pkg/driver/node.go

```diff
@@ -187,6 +186,11 @@ func (d *Driver) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublish
 		return nil, status.Error(codes.InvalidArgument, "Target path not provided")
 	}
 
+	if d.volMetricsOptIn {
+		klog.V(4).Infof("Evicting vol ID: %v, vol path : %v from cache", req.VolumeId, target)
```

Comment:
it's possible to publish the same VolumeId to different targets.

I don't want to complicate this too much, but do we need some kind of ref counter to account for that so we only evict from cache if counter == 0?

Reply:
I think counter will be a bit complex here. Do you know if a volume Id and volume path combination is unique? If yes, I can use it as the cache key

Reply:
It will be unique but if we do that then we will be recalculating the same volume multiple times. Like if I have a volume fs-abcd:/root mounted to both /pod1/target and /pod2/target, the routine will calculate the disk usage of /root in fs-abcd twice, once for /pod1/target and again for /pod2/target which is unnecessary.

Reply:
I now see what you mean with the counter, sure we can add a counter to evict only when all other volumeIds are unpublished.